### DISCUSSION TITLE
Assign project role to user authorization via API

### DISCUSSION THREAD

**Discussion by LuccasGianesini** [zitadel/zitadel#8060] [Q&A]
Hi everyone, i've been trying to add a project role to a user authorization but with no success.

I'm testing Zitadel using the docker compose example (https://zitadel.com/docs/self-hosting/deploy/compose), and i've been trying to add a project role to a user authorization using the api but i had no luck in doing so.  
Using the endpoint `/management/v1/users/:userid/grants` creates a user authorization grant without any roles associated with it, even though the roles where correctly specified in the body.

Is this a bug? Is there any why to set a role to an user authorization grant using the management api?

**Comment by fforootd** [comment]
Hm can you share what your call involves as parameters?

**Comment by LuccasGianesini** [comment]
The  userid is a `ULID` created by the application before it calls the create user endpoint on the zitadel api.

Uri: `/management/v1/users/01HZAVZSYJQ9SPVYDGZ2E8PWAB/grants`
Body:
```json
{
    "projectId": "269294608926638083",
    "roles": [
        "protected:root:99999999"
    ]
}
```

**Reply by hifabienne** [reply]
is the role you are sending "protected:root:99999999" the key you have specified on a role of the project with the id 269294608926638083

**Reply by LuccasGianesini** [reply]
Yes, is the key of one of the project roles

**Reply by LuccasGianesini** [reply]
Hi there, do you have any update on this issue?

**Reply by hifabienne** [reply]
Hi @LuccasGianesini 
Unfortunately, I can't reproduce the problem. Can you once try it through our management console ui, to find out if it works there?
And can you send the complete curl you are doing?

**Reply by LuccasGianesini** [reply]
Hi @hifabienne through the ui it works as expected.

```
curl --location 'http://localhost:8080/management/v1/users/01HZAVZSYJQ9SPVYDGZ2E8PWAB/grants' \
--header 'Content-Type: application/json' \
--header 'Authorization: ••••••' \
--data '{
    "projectId": "269294608926638083",
    "roles": [
        "protected:root:99999999"
    ]
}'
```

**Reply by hifabienne** [reply]
Thanks, I found the problem.
The "roles" in the body is the wrong attribute, it should be roleKeys.
You can find it in our API docs here: https://zitadel.com/docs/apis/resources/mgmt/management-service-add-user-grant

**Reply by LuccasGianesini** [reply]
Awesome, i'll test it today.
Thank you @hifabienne
